l10n_ch_payment_slip/payment_slip.py

```diff
@@ -483,9 +483,6 @@ def _draw_address(self, canvas, print_settings, initial_position, font,
         text.moveCursor(0.0, font.size)
 
         address_lines = com_partner.contact_address.split("\n")
-        if com_partner.country_id:
```

Comment:
This will display the country when we don't want it on the BVR.

For your specific case, you should at least test if country is part of self.country_id.address_format.
Or to make sure it works in any case we should rely on our own address format.


Maybe in an onchange context we could set the com_partner.address_format temporarily.
```
with self.env.do_in_onchange():
    com_partner.country_id.address_format =  "%(street)s\n%(street2)s\n%(city)s %(state_code)s %(zip)s" # or the right format without country_name
   address_lines = com_partner.contact_address.split("\n")
```